test/libdnf/rpm/test_package.cpp

```diff
@@ -0,0 +1,346 @@
+    get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_id();  // Can't really test for the value of the ID
+}
+
+
+void RpmPackageTest::test_get_name() {
+    CPPUNIT_ASSERT_EQUAL(std::string("CQRlib"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_name());
+}
+
+
+void RpmPackageTest::test_get_epoch() {
+    CPPUNIT_ASSERT_EQUAL(0lu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_epoch());
+    CPPUNIT_ASSERT_EQUAL(1lu, get_pkg("nodejs-1:5.12.1-1.fc29.x86_64").get_epoch());
+}
+
+
+void RpmPackageTest::test_get_version() {
+    CPPUNIT_ASSERT_EQUAL(std::string("1.1.1"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_version());
+}
+
+
+void RpmPackageTest::test_get_release() {
+    CPPUNIT_ASSERT_EQUAL(std::string("4.fc29"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_release());
+}
+
+
+void RpmPackageTest::test_get_arch() {
+    CPPUNIT_ASSERT_EQUAL(std::string("x86_64"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_arch());
+}
+
+
+void RpmPackageTest::test_get_evr() {
+    CPPUNIT_ASSERT_EQUAL(std::string("1.1.1-4.fc29"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_evr());
+    CPPUNIT_ASSERT_EQUAL(std::string("1:5.12.1-1.fc29"), get_pkg("nodejs-1:5.12.1-1.fc29.x86_64").get_evr());
+}
+
+
+void RpmPackageTest::test_get_nevra() {
+    CPPUNIT_ASSERT_EQUAL(std::string("CQRlib-1.1.1-4.fc29.x86_64"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_nevra());
+}
+
+
+void RpmPackageTest::test_get_full_nevra() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("CQRlib-0:1.1.1-4.fc29.x86_64"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_full_nevra()
+    );
+}
+
+
+void RpmPackageTest::test_get_group() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("System Environment/Libraries"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_group()
+    );
+}
+
+
+void RpmPackageTest::test_get_size() {
+    CPPUNIT_ASSERT_EQUAL(6324llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_size());
+}
+
+
+void RpmPackageTest::test_get_download_size() {
+    CPPUNIT_ASSERT_EQUAL(6324llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_download_size());
+}
+
+
+void RpmPackageTest::test_get_install_size() {
+    // TODO implement installing packages for tests
+    CPPUNIT_ASSERT_EQUAL(0llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_install_size());
+}
+
+
+void RpmPackageTest::test_get_license() {
+    CPPUNIT_ASSERT_EQUAL(std::string("LGPLv2+"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_license());
+}
+
+
+void RpmPackageTest::test_get_sourcerpm() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("CQRlib-1.1.1-4.fc29.src.rpm"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_sourcerpm()
+    );
+    // TODO is it correct it's empty?
+    CPPUNIT_ASSERT_EQUAL(std::string(), get_pkg("CQRlib-1.1.1-4.fc29.src").get_sourcerpm());
+}
+
+
+void RpmPackageTest::test_get_build_time() {
+    CPPUNIT_ASSERT_EQUAL(1566991198llu, get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_build_time());
+}
+
+
+void RpmPackageTest::test_get_build_host() {
+    // TODO is it correct it's empty?
```

Comment:
To check if it is correct please open metadata and try to find all related information to the package. Libsolv cannot generate data that are not present in xml.

Reply:
Looking into the repo's `primary.xml`, there's `<rpm:buildhost>rohel</rpm:buildhost>`. So that would be non-empty. Looking at definition of `SOLVABLE_BUILDHOST`, which is the libsolv "type" of the build host attribute, it's `"solvable:buildhost"`. I'm not sure what is going on between `primary.xml` and libsolv's loaded repo data though, that would take some time to dig through libsolv. Should I do it or are we happy with the test the way it is for now?

Reply:
I suggest that we should found some package that will return a different value from an empty string. Or do you think that the get_build_host() works incorrectly?

Reply:
I've tried taking a package from the distro (`libdnf`) and call the `get_build_host()` on it, it's empty as well. I don't know whether this is incorrect behavior.

Reply:
I know that it will take a time, but please could you look in libsolv how data are parsed, or if we use the functionality correctly? The method that looks at `SOLVABLE_BUILDHOST` is new in dnf-5, therefore we can also drop it if it is not working correctly.

Reply:
@lukash Please could you report the status?

Reply:
@j-mracek I'm not working on this, as I've said in https://github.com/rpm-software-management/libdnf/pull/1006#issuecomment-672761009 and also on one of our meetings. As I've said, I'd prefer to merge this now and fix the remaining TODOs in a separate PR when I get to it.

Reply:
From comments I am not 100% sure what you found in libsolv. Is functionality to get SOLVABLE_BUILDHOST in libsolv implemented and working? At least I can see that installed RPMs have data about build_host (`rpm -qi dnf`).

Reply:
I didn't look into libsolv at all. Given how complex the internals is I expect it will take some time to find out what's going on. The only information I found is mentioned in https://github.com/rpm-software-management/libdnf/pull/1006#discussion_r466454553

Reply:
Thanks, I just discussed it with dmach and according to his investigation the functionality is not implemented in libsolv. Only flag was created and reserved.

Reply:
Well, I guess that explains it :slightly_smiling_face:

Reply:
It means we cannot merge it like it is.

Reply:
I suggest to create an issue on libsolv, comment out `get_build_host()` from code, add ToDo reference on the issue, and remove `RpmPackageTest::test_get_build_host()`.

Reply:
We can do that. Though the method apparently has been broken in dnf 4 as well and nobody ever minded that. The build host information seems to be practically useless (and anyone needing it can also query the rpm directly). So we could drop it from the API and focus on core functionality.

If we want to pursue this though, I'd keep the method in place, even the test. Create the libsolv issue and when it starts working, the test should actually start failing and thus notify us :slightly_smiling_face:

Reply:
get_build_host() is not in dnf-4 code. I suggest to comment the function out and create an issue. Keeping not functional code in API is dangerous.

Reply:
The function was describe in API-DN5 document created by @dmach. He had a good argument. Why we ship information in metadata/header that we are unable to provide to end/API users.

Reply:
Libsolv issue: https://github.com/openSUSE/libsolv/issues/400